### PR TITLE
CE-3429 + CE-3430 Experiment: editing highlighed text

### PR DESCRIPTION
**Pull request by kamilkoterba**
This is a mobile contribution **experiment** and therefore is not intended to be built in a stable and future-looking way. It is expected that this code be entirely thrown away after the experiment has ended. The experiment will be disabled on March 29.

## Links
https://wikia-inc.atlassian.net/browse/CE-3429
https://wikia-inc.atlassian.net/browse/CE-3430

## Description
This feature notifies user about possibility to edit text that he/she selects and allows to perform the edit.

## Reviewers
@Wikia/community-engineering @Wikia/x-wing

### CODE REVIEW

File: front/main/app/templates/components/article-edit-highlighted-entrypoint.hbs [hunk 1]
```diff
@@ -0,0 +1 @@
+<button class="article-edit-highlighted-button hide-edit" {{action 'editSection'}}>{{svg 'pen' role='img' class='article-edit-highlighted-icon'}}<span>Go to edit mode</span></button>
```
Comment (bognix) [inline front/main/app/templates/components/article-edit-highlighted-entrypoint.hbs]: minor: move the string to i18n files

**Comment by bognix** [comment]
as an experimental code - LGTM

File: front/main/app/components/article-wrapper.js [hunk 2]
```diff
@@ -198,12 +243,18 @@ export default Ember.Component.extend(
 		 * @returns {void}
 		 */
 		didInsertElement() {
+
```
Comment (hakubo) [inline front/main/app/components/article-wrapper.js]: minor: why so many \n?

**Comment by hakubo** [comment]
lgtm - I want to see that already! :)

File: front/main/app/components/article-content.js [hunk 3]
```diff
@@ -540,6 +551,105 @@ export default Ember.Component.extend(
...
+		 * @returns {void}
+		 */
+		launchHighlightedTextEditorDemo() {
+			const highlightedId = 'highlighted-text',
+				$highlightedElement = Ember.$(`#${highlightedId}`),
+				selection = window.getSelection(),
+				range = document.createRange();
+
+			Ember.$(document).unbind('touchmove', this.debouncedScroll);
+			Ember.$(window).unbind('scroll', this.debouncedScroll);
```
Comment (rogatty) [inline front/main/app/components/article-content.js]: These are not unbound if user navigates away from the article before `this.launchHighlightedTextEditorDemo()` is called, right?

You can use `willDestroyElement()` hook to make sure they don't stay forever, see https://guides.emberjs.com/v2.4.0/components/the-component-lifecycle/#toc_detaching-and-tearing-down-component-elements-with-code-willdestroyelement-code

File: front/main/app/components/article-wrapper.js [hunk 4]
```diff
@@ -198,12 +243,18 @@ export default Ember.Component.extend(
...
 			$(window).off('scroll.mercury.preload');
 			window.scrollTo(0, M.prop('scroll'));
 
 			Ember.run.scheduleOnce('afterRender', this, () => {
 				this.sendAction('articleRendered');
 			});
+
+			if (this.get('highlightedEditorEnabled')) {
+				document.addEventListener('selectionchange', this.setHighlightedText.bind(this));
```
Comment (rogatty) [inline front/main/app/components/article-wrapper.js]: What happens after navigating away from article and selecting text? Doesn't it blow up? :) We should probably unbind it on `willDestroyElement()`.

**Comment by rogatty** [comment]
LGTM if it goes together with https://github.com/Wikia/mercury/pull/2129

**Pull request merged**